### PR TITLE
IGNITE-17455 Add setPartition() for IndexQuery

### PR DESCRIPTION
**Pull request by timoninmaxim**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPartitionTest.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    }
+
+    /** */
+    @Test
+    public void testSinglePartition() {
+        int askPart = 23;
+
+        Map<Integer, Person> expRes = new HashMap<>();
+
+        try (IgniteDataStreamer<Integer, Person> dataStreamer = grid(0).dataStreamer("CACHE")) {
```
Comment (ivandasch) [inline modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPartitionTest.java:79]: Actually, the data ingestion code could be exctracted to a separate method returning data distribution.

File: modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPartitionTest.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            grid(0).cache("CACHE").query(qry);
+        },
+            IgniteException.class,
+            "Specified partition must be in the range [0, N) where N is partition number in the cache.");
+    }
+
+    /** */
+    private static class Person {
+        /** */
+        @GridToStringInclude
```
Comment (ivandasch) [inline modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPartitionTest.java:185]: Why is it required to exclude this field from a string representation of the object?

**Review by ivandasch — APPROVED** [review]
Overall, looks good.

I suppose there are some minor issues, but it is up to you whether fix them or not.

File: modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPartitionTest.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+/** */
+public class IndexQueryPartitionTest extends GridCommonAbstractTest {
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        CacheConfiguration<Integer, Person> ccfg = new CacheConfiguration<Integer, Person>()
+            .setName("CACHE")
+            .setAtomicityMode(CacheAtomicityMode.TRANSACTIONAL)
```
[the remark this answers is not in the document]
Comment (timoninmaxim) [reply modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPartitionTest.java:51]: + for testing replicated caches. Added tests.

File: modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPartitionTest.java (hunk 2 above)
Comment (timoninmaxim) [reply modules/indexing/src/test/java/org/apache/ignite/cache/query/IndexQueryPartitionTest.java:185]: It's included (GridToString**Include**) actually.

**Pull request merged**